src/Explorer/Tree/Collection.ts

```diff
@@ -1331,6 +1334,20 @@ export default class Collection implements ViewModels.Collection {
     return this.container.findDatabaseWithId(this.databaseId);
   }
 
+  public async loadAutopilotOfferWithRetry(): Promise<DataModels.Offer> {
+    let retryCount = 0;
+    while (retryCount < Collection.LoadOfferMaxRetries) {
+      if (this.offer().content.offerAutopilotSettings) {
+        return this.offer();
+      } else {
+        await new Promise(resolve => setTimeout(resolve, Collection.LoadOfferRetryAfterMilliSeconds));
```

Comment:
It would be nice to have exponential retries. This doesn't block the main UI thread, right?
What was the issue with p-retry, it seemed like a good idea to not re-invent the wheel? Otherwise, please add unit tests. I would also suggest to make the retry logic re-usable: could be useful somewhere else.